middle_end/flambda/lift_let_to_initialize_symbol.ml

```diff
@@ -43,6 +48,70 @@ type accumulated = {
   terminator : Flambda.expr;
 }
 
+let boxing_closure var kind =
```

Comment:
This might deserve a comment: why we use a closure isn't obvious, it is because it is the only way to make mixed blocks currently, and we need to do that for the upcoming unboxed products to be supported.

Reply:
Done